src/platform/user/profile/utilities/index.js

```diff
@@ -81,8 +81,14 @@ export function setupProfileSession(payload) {
   // this avoids setting the first name to the string 'null'.
   if (firstName) localStorage.setItem('userFirstName', firstName);
 
-  // Report success for the login method.
-  recordEvent({ event: `login-success-${loginPolicy}` });
+  if (sessionStorage.registrationPending) {
```

Comment:
Maybe opt for using `getItem` if possible for consistency. We have to do it for `localStorage` because of our fallback mechanism, but I think we might want to implement a similar fallback for `sessionStorage`.